sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -1914,6 +1914,21 @@ class DataSourceV2SQLSuite
     }
   }
 
+  test("SPARK-30094: current namespace is used during table resolution") {
+    // unset this config to use the default v2 session catalog.
+    spark.conf.unset(V2_SESSION_CATALOG_IMPLEMENTATION.key)
+
+    withTable("spark_catalog.t", "testcat.ns.t") {
+      sql("CREATE TABLE t USING parquet AS SELECT data FROM source")
+      sql("CREATE TABLE testcat.ns.t USING parquet AS SELECT id FROM source")
```

Comment:
can we use different data for these 2 tables? e.g. `CREATE TABLE t ... AS SELECT 1` and `CREATE TABLE testcat.ns.t ... AS SELECT 2`